### PR TITLE
feat: add `productCollectionEntries` method

### PR DESCRIPTION
**Pull request by KrisQ**
## Why are these changes introduced?

Addresses [ENG-8326](https://nacelle.atlassian.net/browse/ENG-8326) 

Adds the `productCollectionEntries` method

## What is this pull request doing?

- Adds the `productCollectionEntries` method to `@nacelle/storefront-sdk-plugin-commerce-queries` to match `@nacelle/storefront-sdk@1.x`'s productCollectionEntries method.

## How to Test

- Check out the `ENG-8178/add-productCollectionEntries-method` branch.
- Navigate to `packages/storefront-sdk-plugins/commerce-queries`.
- npm run coverage - all tests should pass with 100% coverage.
Screen Shot 2023-02-02 at 9 40 21 AM
- `npm run build` - there should be no build errors.

#### Local
- Test locally with `npm link`:
```js
import {StorefrontClient} from '@nacelle/storefront-sdk'
import commerceQueriesPlugin from '@nacelle/storefront-sdk-plugin-commerce-queries'

const storefrontEndpoint = "https://storefront.api.development.nacelle.com/graphql/v1/spaces/<SPACE_ID>"
const ClientWithCommerceQueries = commerceQueriesPlugin(StorefrontClient);
const client = new ClientWithCommerceQueries({ storefrontEndpoint });


const entries = await client.productCollectionEntries({handle: '<HANDLE>'})
```
- Make sure the code works as expected
Screen Shot 2023-02-02 at 9 43 23 AM

## Checklist

- [x] This Pull Request aligns with `nacelle-js`' [Code of Conduct](https://github.com/getnacelle/nacelle-js/blob/main/CODE_OF_CONDUCT.md#code-of-conduct).
- [x] You can follow your own "How to Test" instructions and get the expected result.
- [x] Screenshots, videos, etc. in the PR description are free of brand names and sensitive details.
- [x] Created a [changeset](https://github.com/changesets/changesets) (if the change should appear in a changelog).
- [x] Updated the `README.md` with documentation changes (if applicable).

[ENG-8326]: https://nacelle.atlassian.net/browse/ENG-8326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/storefront-sdk-plugins/commerce-queries/src/index.ts [hunk 1]
```diff
@@ -286,6 +296,102 @@ function commerceQueriesPlugin<TBase extends WithStorefrontQuery & WithConfig>(
...
+								keepFetching = false;
+							}
+						}
+					}
+				}
+			} while (keepFetching);
+
+			return {
+				data: await (this as unknown as StorefrontClient)['applyAfter'](
+					'products',
```
Comment (NWRichmond) [inline packages/storefront-sdk-plugins/commerce-queries/src/index.ts:390]:
```suggestion
					'productCollectionEntries',
```

**Review by NWRichmond — CHANGES_REQUESTED** [review]
The tests are passing with 100% coverage, and everything is working as expected in local testing. Nice!

@nacelle:storefront-sdk-plugin-commerce-queries demo productCollectionEntries

The only reason I'm requesting changes is that the `applyAfter` call for `productCollectionEntries` incorrectly runs for `'products'` method callbacks instead of `productCollectionEntries` method callbacks.

Some smaller things that would be nice to see, but aren't hard requirements:
1. Implementing some/all of the suggested changes in test matchers and GraphQL query name conventions.
2. Some new tests were added in https://github.com/getnacelle/nacelle-js/pull/323/commits/f86d3693ba12f7b8b59cbe1ac4979aa475ffa20b to validate that methods are returning the expected types. Could you please add one for `productCollectionEntries`?
3. nit: the `mockImplementation`/`mockImplementationOnce`s that return a promise could be simplified to `mockResolvedValue`/`mockResolvedValueOnce`. For example:

```ts
// before
mockedFetch.mockImplementation(() => Promise.resolve(thing));

// after
mockedFetch.mockResolvedValue(thing);
```

**Comment by KrisQ** [comment]
Thank you @NWRichmond, good catch on the `applyAfter`! Everything should now be good

**Review by NWRichmond — APPROVED** [review]
While it would still be great to see a test like:

```ts
it('returns data of the expected type', async () => {
	const { data } = await client.productCollectionEntries();

	expectTypeOf(data!).toMatchTypeOf<Product[] | ProductEdge[]>();
});
```

, the bigger things are taken care of. Thanks for that update!

**Comment by KrisQ** [comment]
Good call thanks @NWRichmond, test was added!

File: packages/storefront-sdk-plugins/commerce-queries/src/graphql/documents.ts [hunk 2]
```diff
@@ -2804,6 +3009,184 @@ export const NavigationDocument = {
 		...NavigationItem_NavigationItemFragmentDoc.definitions
 	]
 } as unknown as DocumentNode<NavigationQuery, NavigationQueryVariables>;
+export const ProductCollectionEntriesDocument = {
+	kind: 'Document',
+	definitions: [
+		{
+			kind: 'OperationDefinition',
+			operation: 'query',
+			name: { kind: 'Name', value: 'productCollectionEntries' },
```
Comment (mdarrik) [inline packages/storefront-sdk-plugins/commerce-queries/src/graphql/documents.ts:3018]: Can you run `npm run codegen:all` so this updates this too.

Comment (KrisQ) [reply packages/storefront-sdk-plugins/commerce-queries/src/graphql/documents.ts]: Yes sir! Thank you

**Review by mdarrik — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**